drivers/sensor/qdec_nxp_s32/qdec_nxp_s32.c

```diff
@@ -0,0 +1,429 @@
+			&HwOutput##n##_0_State_Config,						\
+			&HwOutput##n##_1_State_Config,						\
+			&HwOutput##n##_2_State_Config,						\
+			&HwOutput##n##_3_State_Config,						\
+	};											\
+												\
+	const Lcu_Ip_InitType Lcu_Ip_Init_Config##n = {						\
+		.ppxHwOutputStateArray = &Lcu_Ip_ppxHwOutputStateArray##n##_Config[0],		\
+		.ppxLogicInstanceConfigArray = &Lcu_Ip_ppxLogicInstanceArray##n##_Config[0],	\
+		.pxLogicOutputResetConfigArray = &Lcu_Ip_LogicOutputResetConfig##n,		\
+		.pxLogicInputResetConfigArray = &Lcu_Ip_LogicInputResetConfig##n,		\
+		.ppxLogicOutputConfigArray = &Lcu_Ip_ppxLogicOutputArray##n##_Config[0],	\
+		.ppxLogicInputConfigArray = &Lcu_Ip_ppxLogicInputArray##n##_Config[0],		\
+	};
+
+#define Trgmux_Ip_LogicTrigger_Config(n, logic_channel, output, input)				\
+	{											\
+		.LogicChannel = logic_channel,							\
+		.Output = output,								\
+		.Input = input,									\
+		.HwInstId = TRGMUX_NXP_S32_GET_INSTANCE(DT_INST_PHANDLE(n, trgmux)),		\
+		.Lock = (boolean)FALSE,								\
+	};
+
+#define TRGMUX_IP_INIT_CONFIG(n)								\
+	const Trgmux_Ip_LogicTriggerType							\
+		Trgmux_Ip_LogicTrigger##n##_0_Config = Trgmux_Ip_LogicTrigger_Config(n,		\
+		DT_INST_PROP_BY_IDX(n, trgmux_io_config, 0),					\
+		DT_INST_PROP_BY_IDX(n, trgmux_io_config, 1),					\
+		DT_INST_PROP_BY_IDX(n, trgmux_io_config, 2))					\
+	const Trgmux_Ip_LogicTriggerType							\
+		Trgmux_Ip_LogicTrigger##n##_1_Config = Trgmux_Ip_LogicTrigger_Config(n,		\
+			DT_INST_PROP_BY_IDX(n, trgmux_io_config, 3),				\
+			DT_INST_PROP_BY_IDX(n, trgmux_io_config, 4),				\
+			DT_INST_PROP_BY_IDX(n, trgmux_io_config, 5))				\
+	const Trgmux_Ip_LogicTriggerType							\
+		Trgmux_Ip_LogicTrigger##n##_2_Config = Trgmux_Ip_LogicTrigger_Config(n,		\
+			DT_INST_PROP_BY_IDX(n, trgmux_io_config, 6),				\
+			DT_INST_PROP_BY_IDX(n, trgmux_io_config, 7),				\
+			DT_INST_PROP_BY_IDX(n, trgmux_io_config, 8))				\
+	const Trgmux_Ip_LogicTriggerType							\
+		Trgmux_Ip_LogicTrigger##n##_3_Config = Trgmux_Ip_LogicTrigger_Config(n,		\
+			DT_INST_PROP_BY_IDX(n, trgmux_io_config, 9),				\
+			DT_INST_PROP_BY_IDX(n, trgmux_io_config, 10),				\
+			DT_INST_PROP_BY_IDX(n, trgmux_io_config, 11))				\
+	const Trgmux_Ip_InitType Trgmux_Ip_Init_##n##_Config = {				\
+		.paxLogicTrigger = {								\
+			&Trgmux_Ip_LogicTrigger##n##_0_Config,					\
+			&Trgmux_Ip_LogicTrigger##n##_1_Config,					\
+			&Trgmux_Ip_LogicTrigger##n##_2_Config,					\
+			&Trgmux_Ip_LogicTrigger##n##_3_Config,					\
+		},										\
+	};
+
+
+#define QDEC_NXP_S32_INIT(n)									\
+												\
+	static struct qdec_s32_data qdec_s32_##n##_data = {					\
+		.micro_ticks_per_rev = (float)(DT_INST_PROP(n, micro_ticks_per_rev) / 1000000), \
+		.counter_CW = 1,								\
+		.counter_CCW = 1,								\
+	};											\
+												\
+	PINCTRL_DT_INST_DEFINE(n);								\
+	TRGMUX_IP_INIT_CONFIG(n)								\
+	LCU_IP_INIT_CONFIG(n)									\
```

Comment:
Not a change request on this PR if you plan to address this in a follow up PR, but beware that the instance `n` refers to the QDEC instance (which is a software instance because there is no QDEC hw peripheral), but is also used to configure the hw instance `LCU_NXP_S32_GET_INSTANCE()`. The same applies for TRGMUX. The only way to solve this is to separate TRGMUX/LCU into their own drivers so they can keep their own private data for the respective hw instance as I initially mentioned.